### PR TITLE
ICU-22406 Fix test breakage w/ --disabled-shared

### PR DESCRIPTION
**Pull request by FrankYFTang**
1. Added github action to test against build with 
  a. --enable-static, AND
  b. --enable-static --disable-shared
2. Fix the issue which the test/testdata *.res build w/o ucadata.icu by using -i with the build directory..

##### Checklist

- [X] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22406
- [X] Required: The PR title must be prefixed with a JIRA Issue number. 
- [X] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [X] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [X] Issue accepted (done by Technical Committee after discussion)
- [X] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: icu4c/source/test/testdata/BUILDRULES.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def generate_rb(config, io, common_vars):
             input_files = [InFile("%s.txt" % bn) for bn in basenames],
             output_files = [OutFile("%s.res" % bn) for bn in basenames],
             tool = IcuTool("genrb"),
-            args = "-q -s {IN_DIR} -d {OUT_DIR} {INPUT_FILE}",
+            args = "-q -i {OUT_DIR}/../../../../data/out/build -s {IN_DIR} -d {OUT_DIR} {INPUT_FILE}",
```
Comment (sffc) [inline icu4c/source/test/testdata/BUILDRULES.py:66]: This is probably going to be brittle. If we need to know the path to the non-test data files while running genrb for test data files, we should provide that path in the environment. Here's how you would plumb it through:

1. Pick a name for the variable; something like `LIBRARY_DATA_DIR`
2. Calculate the path in `icu4c/source/test/testdata/Makefile.in` and set it to the variable; see "Variable names for rules.mk" in that file
3. Edit `icu4c/source/python/icutools/databuilder/__main__.py` to plumb that variable into the `makefile_env`. I'm not sure about running the tool in non-Makefile mode; presumably you have the same problem, so you may need to add a flag to `databuilder` like we do for all the other paths.
4. Now you can use that variable in this args path: `{LIBRARY_DATA_DIR}`

File: .github/workflows/icu_ci.yml [hunk 2]
```diff
@@ -250,6 +250,29 @@ jobs:
...
           make dist
 
+  # clang build with some options
+  clang-options-build-and-test:
+    strategy:
+      # "fail-fast: false" lets other jobs keep running even if the test breaks in some other options.
+      fail-fast: false
+      matrix:
+        build_option:
+         [ --enable-static, --enable-static --disable-shared ]
```
Comment (sffc) [inline .github/workflows/icu_ci.yml:260]: My guess is that what's _probably_ going on here is that genrb runs with an LD_LIBRARY_PATH, which normally should probably include *libicudata.so*, but now we don't have that shared library any more, so we can't find the data.

Comment (FrankYFTang) [reply .github/workflows/icu_ci.yml:260]: The bug is under the configuration of "--disable-shared" so regardless we have  libicudata.so, genrb will not include it.

**Comment by FrankYFTang** [comment]
PTAL. not sure is this what you mean

File: icu4c/source/test/testdata/Makefile.in [hunk 3]
```diff
@@ -92,6 +92,7 @@ TESTSRCDATADIR=$(top_srcdir)/test/testdata
 TESTOUTDIR=$(top_builddir)/test/testdata/out
 BUILD_DIRS = $(TESTOUTDIR) $(TESTBUILDDIR) $(TESTOUTDIR)/$(TESTDT)
 GENTEST=$(TOOLDIR)/gentest/gentest$(TOOLEXEEXT)
+LIBRARY_DATA_DIR=$(OUTDIR)/build
 
 ifeq ($(PKGDATA_MODE),common)
 ICU_DATA_OPT = -i $(OUTDIR)
```
Comment (sffc) [inline icu4c/source/test/testdata/Makefile.in:98]: I just noticed that there is already this `ICU_DATA_OPT` being set, but I don't know if it's being used anywhere. Hmm.

**Review by sffc — APPROVED** [review]
Seems good if it works.

**Pull request merged**